CMakeLists.txt

```diff
@@ -0,0 +1,191 @@
+cmake_minimum_required(VERSION 3.12)
+
+option(BUILD_TESTS "Build test executables" OFF)
+option(BUILD_DOCS "Build documentation" OFF)
+option(HUNTER_ENABLED "Use the Hunter package manager" OFF)
+option(BUILD_FOR_LIBKTX "Build dfdutils for use in KTX-Software" OFF)
+
+if(HUNTER_ENABLED)
+    include(cmake/HunterGate.cmake)
+    HunterGate(
+        URL "https://github.com/cpp-pm/hunter/archive/v0.23.309.tar.gz"
+        SHA1 "15e84e1f23eb079360db727d373dab82890420b7"
+    )
+endif()
+
+project(dfdutils VERSION 0.0.0)
+set(VULKAN_HEADERS_VERSION "1.2.182" CACHE STRING "Version of Vulkan Headers to use (patch in third_party/ must apply).")
+
+if(HUNTER_ENABLED)
+    hunter_add_package(KhronosDataFormat)
+    find_package(KhronosDataFormat CONFIG REQUIRED)
+
+    hunter_add_package(Vulkan-Headers)
+    find_package(VulkanHeaders CONFIG REQUIRED)
+
+    get_target_property(DataFormat_INCLUDE_DIR Khronos::KhronosDataFormat INTERFACE_INCLUDE_DIRECTORIES)
+    get_target_property(Vulkan_INCLUDE_DIR Vulkan::Headers INTERFACE_INCLUDE_DIRECTORIES)
+else()
+    include(FetchContent)
+    FetchContent_Declare(
+        khronosdataformat
+        URL https://github.com/KhronosGroup/DataFormat/archive/1f8c8529b712e4b2b17bde798636d9ea550af3ed.tar.gz
+    )
+    FetchContent_Populate(khronosdataformat)
+    file(INSTALL "${khronosdataformat_SOURCE_DIR}/headers/khr_df.h" DESTINATION "${CMAKE_CURRENT_BINARY_DIR}/third_party/KHR")
+    set(DataFormat_INCLUDE_DIR "${CMAKE_CURRENT_BINARY_DIR}/third_party")
+
+    FetchContent_Declare(
+        vulkanheaders
+        URL https://github.com/KhronosGroup/Vulkan-Headers/archive/refs/tags/v${VULKAN_HEADERS_VERSION}.tar.gz
+    )
+    FetchContent_Populate(vulkanheaders)
+    file(INSTALL "${vulkanheaders_SOURCE_DIR}/include/vulkan" DESTINATION "${CMAKE_CURRENT_BINARY_DIR}/third_party")
+    set(Vulkan_INCLUDE_DIR "${CMAKE_CURRENT_BINARY_DIR}/third_party")
+endif()
+
+set(GENERATED_DIR "${CMAKE_CURRENT_BINARY_DIR}/generated")
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake/modules/")
+
+include(cmake/mkvk.cmake)
+
+add_library(dfdutils STATIC)
+
+set(DFDUTILS_HDR_FILES
+    dfd.h
+    ${GENERATED_DIR}/vkformat_enum.h
+)
+
+set(DFDUTILS_SRC_FILES
+    colourspaces.c
+    createdfd.c
+    ${GENERATED_DIR}/dfd2vk.inl
+    interpretdfd.c
+    printdfd.c
+    queries.c
+    vk2dfd.c
+    ${GENERATED_DIR}/vk2dfd.inl
+    ${GENERATED_DIR}/vkformat_check.c
+    ${GENERATED_DIR}/vkformat_str.c
```

Comment:
These 2, especially vkformat_check.c, are not truly part of the dfdutils. As I mentioned elsewhere I need to refresh my memory re. the forbidden formats to see if a similar file is needed for dfdutils. I think not because vk2dfd.inl will reflect any unsupported formats. KTX-Software uses the file to check vkformat values in incoming .KTX2 files.

Reply:
I've checked the KTX v2 prohibited formats. There are 3 classes of them. The first 2 classes should all be handled by vk2dfd. The 3rd class is multisample formats which are not supported by KTX but can be described by a DFD. I think vk2dfd probably doesn't handle them at present but it could. vkformat_check.c prohibits all 3 classes so it does not belong in dfdutils. Leave it to KTX-Software.

I am debating with myself about vkformat_str. An error message using a format string would be emitted by a caller of dfdutils not by anything in dfdutils which suggests removing it too. But having it here removes the need for the each caller to create something similar so I can see reason to leave it here. But they are both generated by the same awk script so I think it best to remove vkformat_str too.